scheduler/utilization_based_host_allocator.go

```diff
@@ -0,0 +1,248 @@
+package scheduler
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/model/distro"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+)
+
+func UtilizationBasedHostAllocator(ctx context.Context, hostAllocatorData HostAllocatorData) (map[string]int, error) {
+	count := 0
+	var distroName string
+	newHostsNeeded := make(map[string]int)
+
+	// once the input format changes this loop can be removed - should only have 1 distro
```

Comment:
could we make this change separately/now?

Reply:
We should do it separate from this since it changes the interface for all the host allocators. I'll open a ticket